content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,268 @@
+  var enc = new TextDecoder();
+  var channel = ably.channels.get('input');
+  channel.subscribe(function(message) {
+    var command = enc.decode(message.data);
+    console.log(command);
+  });
+```
+
+This will simply subscribe to the @input@ channel you're publishing to from the controller, and print it into the console. The @TextDecoder@ is required due to MQTT being a binary protocol with no encoding. This means the payload you receive is just raw bits that need to be interpreted. In this case we're interpreting it as text with the @TextDecoder@.
+
+Open up @snake.html@ in a browser, open up the browser's "console":http://wickedlysmart.com/hfjsconsole/, and run your controller by typing @node controller.js@ into your command line. Now when you press the arrow keys from your controller, you should see messages being received by the browser in the console.
+
+<a href="/images/tutorials/mqtt-snake-keys-received.png" target="_blank">
+  <img src="/images/tutorials/mqtt-snake-keys-received.png" style="width: 100%" alt="Sending messages through Ably with MQTT">
+</a>
+
+"See this step in Github":https://github.com/ably/tutorials/commit/c325a1c4f161f5c49bbf7ee6cec280837a8749fe
+
+h2. Step 8 - Play Snake!
+
+With the setup complete, you can now add in the code for the actual snake game. Replace the contents of @snake.js@ with "this":https://github.com/ably/tutorials/commit/e8379c790103bcc3984dc068ea252f538abdf523 code, replacing REPLACE_WITH_YOUR_API_KEY with your API Key. When you reload the @snake.html@ page you should see your basic Snake game. Run the controller as before, and you can now play! Press enter to start, arrow keys to move, and try to grow the Snake as large as you can.
+
+<a href="/images/tutorials/mqtt-snake-playing.gif" target="_blank">
+  <img src="/images/tutorials/mqtt-snake-playing.gif" style="width: 100%" alt="Playing Snake!">
+</a>
+
+The complete source code for each step of "this tutorial is available on Github":https://github.com/ably/tutorials/commits/mqtt-snake-tutorial.
+
+We recommend that you clone the repo locally:
+
+bc[sh]. git clone https://github.com/ably/tutorials.git
+
+Checkout to the tutorial branch:
+
+bc[sh]. git checkout mqtt-snake-tutorial
+
+And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @snake.js@ and @controller.js@
+
+h2. Next steps
+
+1. Learn more about our "support of other protocols":https://www.ably.io/adapters
+2. Learn more about "Ably features":https://www.ably.io/features by going through our other "Ably tutorials":https://www.ably.io/tutorials
+3. Gain a good technical "overview of how the Ably realtime platform works":https://www.ably.io/documentation/how-ably-works
+4. "Get in touch if you need help":https://www.ably.io/contact
+
+5. Created something cool based on this tutorial? "Tweet it to us!":https://twitter.com/ablyrealtime
```

Comment:
Why the extra space?

Reply:
This was per @Srushtika's feedback suggesting we try to make it stand out by separating it. I'm happy to do whatever in regards to this however.

Reply:
Hey @tomczoink, my apologies if my comment wansn't clear. I meant more like not keeping it in the list as the fifth item but putting it out in a new paragraph/ heading etc :)

Reply:
Ah sure thing then, I'll change it!